controllers/fluxcd/application-controller.go

```diff
@@ -268,12 +295,14 @@ func (r *ApplicationReconciler) updateKustomization(ctx context.Context, app *v1
 }
 
 func isHelmOrKustomize(app *v1alpha1.Application) (AppType, error) {
-	conf := app.Spec.FluxApp.Spec.Config.DeepCopy()
-	helm, kus := conf.HelmRelease != nil, conf.Kustomization != nil
-	if helm && !kus {
+	if app.Spec.FluxApp.Spec.Config == nil {
```

Comment:
This inspection method has appeared many times, how about change it to a function